iree/turbine/kernel/wave/wave.py

```diff
@@ -583,17 +584,26 @@ def _trace_and_get_kernel_signature(
 
         # Add grid and block dims to kernel launch info.
         # Convert the grid into a lambda that we can use to compute the grid dimension.
+        hw_constraint = get_hardware_constraint(self.constraints)
         options.kernel_launch_info.grid = sympy.lambdify(
             [list(options.dynamic_symbols_map.keys())], self.grid_type.dims
         )
         options.kernel_launch_info.grid_str = lambdastr(
             [list(options.dynamic_symbols_map.keys())], self.grid_type.dims
         )
         options.kernel_launch_info.blocks = [
-            int(x) for x in get_hardware_constraint(self.constraints).threads_per_block
+            int(x) for x in hw_constraint.threads_per_block
         ]
         options.kernel_launch_info.func_name = self._name
 
+        idxc = IndexingContext.current()
+        for sym, val in zip(
+            [THREAD_0, THREAD_1, THREAD_2, WORKGROUP_0, WORKGROUP_1, WORKGROUP_2],
+            chain(hw_constraint.threads_per_block, self.grid_type.dims),
```

Comment:
Chain here is combining/concatenating two range iterators (if list it's just a +)